lemur/certificates/models.py

```diff
@@ -242,7 +245,15 @@ def __init__(self, **kwargs):
         self.dns_provider_id = kwargs.get("dns_provider_id")
 
         for domain in defaults.domains(cert):
-            self.domains.append(Domain(name=domain))
+            # Avoid circular import.
+            from lemur.domains import service as domain_service
+
+            # Create an association with an existing Domain, otherwise, create a new Domain.
+            existing_domain = domain_service.get_by_name_sorted_by_id(domain, "asc")
```

Comment:
Would it be helpful to add a limit to the query to avoid getting potentially thousands of results back here (if the migration hasn't been run)?

Reply:
Since this is intended to get domain with lowest id for the matching name, it can fetch only first record than fetching all and using `existing_domain[0]`
e.g. `return database.get_all_sorted(Domain, name, field="name", sort_by="id", sort_dir=sort_dir).first()`
But to Jasmine's point, does this run in a reasonable time for a large dataset if migration hasn't been run?

Reply:
Search by domain is an indexed search, so should be fast.  However, there may be many duplicated records returned (if db migration hasn't been done), so returning just one is probably preferred here.